modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/AssignQualityScore.pm

```diff
@@ -59,30 +59,41 @@ sub fetch_input {
 	foreach my $oid ( @orth_ids ){
 		$sth->execute($oid);
 		$max_quality{$oid} = $sth->fetchrow_arrayref->[0] or $self->warning("Cannot find quality scores in db for homology id $oid");
-	}
+        $max_quality{$oid} = 0 unless $max_quality{$oid}; # default to 0 like GOC
+    }
 	
 	$self->param('max_quality', \%max_quality);
 }
 
 =head2 write_output
 
-	Description: write avg score to homology table & threshold to mlss_tag
+Description: write avg score to file
 
 =cut
 
 sub write_output {
-	my $self = shift;
+    my $self = shift;
 
-        $self->dbc->disconnect_if_idle() if $self->dbc;
+    # disconnect from dbs
+    $self->compara_dba->dbc->disconnect_if_idle();
+    $self->dbc->disconnect_if_idle() if $self->dbc;
+    $self->data_dbc->disconnect_if_idle();
 
-	my $homology_adaptor = $self->compara_dba->get_HomologyAdaptor;
-	my %max_quality      = %{ $self->param('max_quality') };
-	foreach my $oid ( keys %max_quality ) {
-		$homology_adaptor->update_wga_coverage( $oid, $max_quality{$oid} );
-	} 
+    my $output_file = $self->param('output_file');
+    my $reuse_file = $self->param('reuse_file');
+    $self->run_command("mkdir -p " . dirname($output_file)) unless -d dirname($output_file);
+    $self->run_command("cp $reuse_file $output_file") if ( -e $reuse_file );
```

Comment:
If `$reuse_file` is not defined, we should be resetting / emptying the output file. Can be done here or below by using `>` instead of `>>` ?